packages/form/addon/lib/document.js

```diff
@@ -209,6 +210,16 @@ export default class Document extends Base {
             form: this.rootForm.slug,
             formMeta: this.rootForm.raw.meta,
           },
+          case: _case
+            ? {
+                form: _case.document?.form.id,
+                workflow: _case.workflow.id,
+                root: {
+                  form: _case.family.document?.form.id,
+                  workflow: _case.family.workflow.id,
+                },
+              }
+            : null,
```

Comment:
```suggestion
          case: {
                form: _case?.document?.form.id,
                workflow: _case?.workflow.id,
                root: {
                  form: _case?.family.document?.form.id,
                  workflow: _case?.family.workflow.id,
                },
              }
```

Reply:
The previous implementation also worked for unlinked table rows, but I refactored it as suggested anyway.